pkg/agent/svid/rotator_test.go

```diff
@@ -169,37 +301,72 @@ func (c *fakeClient) RenewSVID(ctx context.Context, csrBytes []byte) (*client.X5
 		return nil, err
 	}
 
-	tmpl := &x509.Certificate{
-		SerialNumber: big.NewInt(1),
-		NotBefore:    c.clk.Now(),
-		NotAfter:     c.clk.Now().Add(time.Hour),
-	}
-
-	svid, err := x509.CreateCertificate(rand.Reader, tmpl, c.caCert, csr.PublicKey, c.caKey)
+	notAfter := c.clk.Now().Add(time.Hour)
+	svid, err := createSVID(csr.PublicKey, c.caCert, c.caKey, c.clk.Now(), notAfter)
 	if err != nil {
 		return nil, err
 	}
 
 	return &client.X509SVID{
 		CertChain: svid,
-		ExpiresAt: tmpl.NotAfter.Unix(),
+		ExpiresAt: notAfter.Unix(),
 	}, nil
 }
 
 func (c *fakeClient) Release() {
 	c.releaseCount++
 }
 
-func createTestSVID(t *testing.T, svidKey crypto.Signer, ca *x509.Certificate, caKey crypto.Signer, notBefore time.Time, notAfter time.Time) []*x509.Certificate {
+type fakeAgentService struct {
+	clk          clock.Clock
+	attested     bool
+	svidKM       keymanager.SVIDKeyManager
+	svidKey      keymanager.Key
+	caCert       *x509.Certificate
+	caKey        crypto.Signer
+	failReattest bool
+
+	agentv1.AgentServer
```

Comment:
nit: can you move `agentv1.AgentServer` at the start of struct?

Reply:
done